### PR TITLE
Use polymorphism for file content matcher

### PR DESCRIPTION
**Pull request by mattwynne**

### CODE REVIEW

File: lib/aruba/matchers/file.rb [hunk 1]
```diff
@@ -63,34 +63,96 @@
...
-#     end
-#
-#   @example Use matcher with regexp
-#
-#     RSpec.describe do
-#       it { expect(file1).to have_file_content(/a/) }
-#     end
+module RSpec
+  module Matchers
+    # @!method have_file_content(content)
```
Comment (dg-ratiodata) [inline lib/aruba/matchers/file.rb]: I think you can remove `@!method ...`.

**Comment by dg-ratiodata** [comment]
I like the idea a lot. :smile:

File: lib/aruba/matchers/file.rb [hunk 2]
```diff
@@ -63,34 +63,96 @@
...
+    #       it { expect(file1).to have_file_content('a') }
+    #     end
+    #
+    #   @example Use matcher with regexp
+    #
+    #     RSpec.describe do
+    #       it { expect(file1).to have_file_content(/a/) }
+    #     end
+    def have_file_content(expected)
+      if expected.is_a? Regexp
```
Comment (dg-ratiodata) [inline lib/aruba/matchers/file.rb]: I think we could refactor this to something like this

```ruby
@matchers = []
@matchers <<  Aruba::Matchers::HaveExactFileContent
@matchers << Aruba::Matchers::HaveMatchingFileContent

class Aruba::Matchers::HaveMatchingFileContent
  def self.handles?(input)
    input.is_a? Regexp
  end
end

# Handles everything else => always true
class Aruba::Matchers::HaveExactFileContent
  def self.handles?(input)
    true
  end
end

@matchers.find { |m| m.handles? expected }
```

**Comment by dg-ratiodata** [comment]
Should we use this kind of pattern for all other matchers as well? Or should we use the DSL for simpler ones?

**Comment by dg-ratiodata** [comment]
@jarl-dk @mattwynne 
Opinions to that?

**Comment by myronmarston** [comment]
RSpec 3.x provides much improved support for composable matchers that means you don't have to worry about string vs regexp.  I'll try opening a PR showing what I mean in a bit.

File: lib/aruba/matchers/file.rb (hunk 2 above)
Comment (mattwynne) [inline lib/aruba/matchers/file.rb]: very nice!

Comment (mattwynne) [inline lib/aruba/matchers/file.rb]: remember you can just push commits to this branch now you've got the karma

**Comment by mattwynne** [comment]
No need to mention me @dg-ratiodata, I get notified of each comment in this repo anyway.

Let's see what @myronmarston comes up with, thanks Myron!

**Comment by myronmarston** [comment]
See #249.

**Comment by jarl-dk** [comment]
This looks cool, though superseded by #249 IMHO. The PR needs som rubocop work, but then again if we are heading towards #249 maybe just leave this.

**Comment by mattwynne** [comment]
Indeed, #249 is the one.

**Pull request closed without merging**